backend/src/controllers/profileController.js

```diff
@@ -0,0 +1,48 @@
+import uuidv4 from '../helpers/uuid.js';
+import { createProfileSchema, getProfileSchema } from '../helpers/schemas.js';
+
+
+export default class ProfileController {
+    static profiles = []; // TODO: replace with database
+
+    static getProfiles(req, res) {
+        res.json(ProfileController.profiles);
+    }
+
+
+    static getProfile(req, res) {
+        try {getProfileSchema.isValid(req.params.id).then((isValid) => console.log(`is the profile valid? ${isValid}`));
+
+            getProfileSchema.validateSync(req.params.id);
+
+            const userId = req.params.id;
+            const user = ProfileController.profiles.find(user => user.id === userId);
+
+            if (user) {
```

Comment:
Also I suggest to revert this condition. Much easier to read if it was
if (!user) res.status(404).json({ message: `User with id = ${userId} not found :(` });
.... rest of your code here
So by that you'll eliminate this extra `else` statement (and improves code readability)